crema/qvalues.py

```diff
@@ -132,3 +136,213 @@ def _fdr2qvalue(scores, fdr):
         start = idx + 1
 
     return qvals
+
+
+def mixmax(target_scores, decoy_scores, combined_score, combined_score_target):
+    """
+    Estimate q-values using mix-max competition.
+
+    TODO
+    Estimates q-values using...
+
+    Parameters
+    ----------
+    target_scores : numpy.ndarray of float
+       An array of the best target PSM score per spectrum.
+    decoy_scores : numpy.ndarray of float
+       An array of the best decoy PSM score per spectrum.
+    combined_score : pandas.DataFrame
+       A :py:class:`pandas.DataFram` of PSMs. Dataframe includes
+       best ranked target and decoy per spectrum.
+    combined_score_target: numpy.ndarray of float
+       The target/deoy column from the combined_score dataframe
+
+    Returns
+    ----------
+    pi0 : float
+        Estimated pi_zero.
+    fdrmod : numpy.ndarray
+        Array of calculated q-values.
+    """
+    # TODO try except and some error checking
+
+    num_targets = target_scores.shape[0]
+    num_decoys = decoy_scores.shape[0]
+
+    # calculate p-values from scores
+    nDecoys = 1
+    posSame = 0
+    negSame = 0
+    pValList = []
+    curScore = None
+    for score, target in zip(combined_score, combined_score_target):
+        if target:
+            posSame += 1
+        else:
+            negSame += 1
+
+        # TODO Unclear if need if statement. This appears in Percolator code
+        # however, adding it makes Crema and Percolator pValList different length
+        # if curScore != score:
+        for ix in range(0, posSame):
+            pValList.append(nDecoys + (negSame * (ix + 1)) / (posSame + 1))
+
+        nDecoys += negSame
+        negSame = 0
+        posSame = 0
+        curScore = score
+    pValList = np.array(pValList) / nDecoys
+
+    # calculate pi0
+    pi0 = estimate_pi0(pValList)
+
+    if pi0 == 1.0:
+        # All targets are assumed to be incorrect! Algorithm 1, line 5-6
+        LOGGER.debug("FALLBACK: pi0==1.0; all q-values will be 1.0")
+        fdrmod = np.full(num_targets, 1.0)  # all q-values are 1
+    elif pi0 < 0 or pi0 >= 1 or not np.isfinite(pi0):
+        raise ValueError(
+            f"Invalid pi0 estimate ({pi0}); unable to proceed FDR estimation!"
+        )
+    else:
+        fdrmod = calculate_mixmax_qval(
+            np.array(target_scores), np.array(decoy_scores), pi0
+        )
+
+    return (pi0, fdrmod)
```

Comment:
**nitpick:** The variables in this function should be converted from `camelCase` to `snake_case`.